src/Makefile

```diff
@@ -301,6 +301,17 @@ ifeq ($(BUILD_TLS),ext)
 	REDIS_TLS_EXT_CFLAGS+=-DUSE_OPENSSL=$(BUILD_YES) $(OPENSSL_CFLAGS) -DBUILD_TLS_EXT
 endif
 
+REDIS_UNIX_EXT=
+REDIS_UNIX_EXT_NAME:=redis-unix$(PROG_SUFFIX).so
+REDIS_UNIX_EXT_CFLAGS:=$(FINAL_CFLAGS)
+ifeq ($(BUILD_UNIX),ext)
+	FINAL_CFLAGS+=-DUSE_UNIX=$(BUILD_EXT)
+	REDIS_UNIX_EXT=$(REDIS_UNIX_EXT_NAME)
+	REDIS_UNIX_EXT_CFLAGS+=-DUSE_UNIX=$(BUILD_YES) -DBUILD_UNIX_EXT
+else
+	FINAL_CFLAGS+=-DUSE_UNIX=$(BUILD_YES)
+endif
```

Comment:
it's a nice proof of concept, but why would we want to load the Unix socket as an extension, or exclude it completely?
unlike the TLS thing, it doesn't require any heavy dependencies or overheads..
i think i'd like to drop that commit.